python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -0,0 +1,496 @@
+
+
+class LightCurveGenerator(object):
+    """
+    This class will find all of the OpSim pointings in a particular region
+    of the sky in a particular filter and then return light curves for all
+    of the objects observed in that region of sky.
+
+    Input parameters:
+    -----------------
+    catalogdb is a CatalogDBObject instantiation connecting to the database
+    of objects to be observed.
+
+    opsimdb is the path to the OpSim database of observation.
+
+    opsimdriver (optional; default 'sqlite') indicates the database driver to
+    be used when connecting to opsimdb.
+    """
+
+    _lightCurveCatalogClass = None
+
+    def __init__(self, catalogdb, opsimdb, opsimdriver="sqlite"):
+        self._generator = ObservationMetaDataGenerator(database=opsimdb,
+                                                       driver=opsimdriver)
+
+        self._catalogdb = catalogdb
+
+    def _filter_chunk(self, chunk):
+        return chunk
+
+    def get_pointings(self, ra, dec, bandpass=('u', 'g', 'r', 'i', 'z', 'y'), expMJD=None):
+        """
+        Inputs
+        -------
+        ra is a tuple indicating the (min, max) values of RA in degrees.
+
+        dec is a tuple indicating the (min, max) values of Dec in degrees.
+
+        bandpass is a str (i.e. 'u', 'g', 'r', etc.) or an iterable indicating
+        which filter(s) you want the light curves in.  Defaults to all six
+        LSST bandpasses.
+
+        expMJD is an optional tuple indicating a (min, max) range in MJD.
+        Defaults to None, in which case, the light curves over the entire
+        10 year survey are returned.
+
+        Outputs
+        -------
+        A 2-D list of ObservationMetaData objects.  Each row is a list of
+        ObservationMetaDatas that point to the same patch of sky, sorted by MJD.
+        Pointings will not be sorted or grouped by filter.
+        """
+
+        print('parameters', ra, dec, bandpass, expMJD)
+        if isinstance(bandpass, str):
+            obs_list = self._generator.getObservationMetaData(fieldRA=ra,
+                                                              fieldDec=dec,
+                                                              telescopeFilter=bandpass,
+                                                              expMJD=expMJD,
+                                                              boundLength=1.75)
+        else:
+            obs_list = []
+            for bp in bandpass:
+                sub_list = self._generator.getObservationMetaData(fieldRA=ra,
+                                                                  fieldDec=dec,
+                                                                  telescopeFilter=bp,
+                                                                  expMJD=expMJD,
+                                                                  boundLength=1.75)
+                obs_list += sub_list
+
```

Comment:
Would be nice to add a comment saying obs_list is a list of ObservationMetaData Instances with pointingRA, pointingDec, filters, observed MJD values in the ranges indicated by the tuples ra, dec, expMJD and bp